File: VB/Busy/BusyForm.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace Busy

    Public Partial Class BusyForm
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub BusyForm_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs)
            If realProgress1.IsOperationInProgress Then
                e.Cancel = True
                MessageBox.Show(Me, "This operation in not completed yet. Please wait.", "Cannot close this form")
            End If
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Text += "Click!"
        End Sub
    End Class
End Namespace

File: CS/Busy/BusyForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace Busy {
    public partial class BusyForm : Form {
        public BusyForm() {
            InitializeComponent();
        }

        private void BusyForm_FormClosing(object sender, FormClosingEventArgs e) {
            if(realProgress1.IsOperationInProgress) {

                e.Cancel = true;
                MessageBox.Show(this, "This operation in not completed yet. Please wait.", "Cannot close this form");
            }
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            Text += "Click!";
        }
    }
}
